src/main/scala/vsys/state/opcdiffs/LoadOpcDiff.scala

```diff
@@ -0,0 +1,39 @@
+package vsys.state.opcdiffs
+
+import scorex.transaction.ValidationError
+import scorex.transaction.ValidationError.{ContractInvalidOPCData, ContractLocalVariableIndexOutOfRange}
+import vsys.contract.{DataEntry, DataType, ExecutionContext}
+
+import scala.util.{Left, Right}
+
+object LoadOpcDiff {
+
+  def signer(context: ExecutionContext)(dataStack: Seq[DataEntry], pointer: Byte): Either[ValidationError, Seq[DataEntry]] = {
+    if (pointer > dataStack.length || pointer < 0) {
+      Left(ContractLocalVariableIndexOutOfRange)
+    } else {
+      Right(dataStack.patch(pointer, Seq(DataEntry(context.signers.head.bytes.arr, DataType.Address)), 1))
+    }
+  }
+
+  def caller(context: ExecutionContext)(dataStack: Seq[DataEntry], pointer: Byte): Either[ValidationError, Seq[DataEntry]] = {
```

Comment:
caller() seems exactly the same as signer().

Reply:
yes, this is same as isSignerOrigin and isCallerOrigin